In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os


%matplotlib inline

# Limit rows disp# Limit rows displayed in notebook
pd.set_option('display.max_rows', 20)
pd.set_option('display.precision', 2)

In [4]:
# reviews
star_1 = pd.read_pickle('by_stars/1reviews10k_with_topics.pkl')
star_2 = pd.read_pickle('by_stars/2reviews10k_with_topics.pkl')
star_3 = pd.read_pickle('by_stars/3reviews10k_with_topics.pkl')
star_4 = pd.read_pickle('by_stars/4reviews10k_with_topics.pkl')
star_5 = pd.read_pickle('by_stars/5reviews10k_with_topics.pkl')

In [5]:
star_1_topics = pd.read_pickle('by_stars/10ktopics_star_1.pkl')
star_2_topics = pd.read_pickle('by_stars/10ktopics_star_2.pkl')
star_3_topics = pd.read_pickle('by_stars/10ktopics_star_3.pkl')
star_4_topics = pd.read_pickle('by_stars/10ktopics_star_4.pkl')
star_5_topics = pd.read_pickle('by_stars/10ktopics_star_5.pkl')

In [6]:
restaurants = pd.read_pickle('restaurant/final_restaurant_dataset.pkl')
print restaurants.describe()
restaurants.head()

         open  review_count   stars  cat.Active Life  cat.Adult Entertainment  \
count  3358.0        3358.0  3358.0          3.4e+03                     3358   
mean      0.9          85.5     3.6          2.4e-03                        0   
std       0.3          72.5     0.6          4.9e-02                        0   
min       0.0           3.0     1.5          0.0e+00                        0   
25%       1.0          34.0     3.5          0.0e+00                        0   
50%       1.0          62.0     3.5          0.0e+00                        0   
75%       1.0         113.0     4.0          0.0e+00                        0   
max       1.0         384.0     5.0          1.0e+00                        0   

       cat.Afghan  cat.African  cat.Airports  cat.American (New)  \
count     3.4e+03      3.4e+03          3358              3358.0   
mean      8.9e-04      6.0e-04             0                 0.1   
std       3.0e-02      2.4e-02             0                 0.3  

business_id  open  review_count  stars  cat.Active Life  \
0  EmzaQR5hQlF0WIl24NxAZA     1            23    2.5                0   
1  KPoTixdjoJxSqRSEApSAGg     1           322    4.0                0   
2  e5kc0CQ4R-PCCDgb274gSg     1           191    3.5                0   
3  45puCRQ6Vh_IIAy7kkfFDQ     1           170    3.5                0   
4  BESmH-lmMm34_EefL8X9fQ     1           204    4.0                0   

   cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                        0           0            0             0   
1                        0           0            0             0   
2                        0           0            0             0   
3                        0           0            0             0   
4                        0           0            0             0   

   cat.American (New)    ...      Noise.average  Noise.loud  Noise.quiet  \
0                   1    ...                  0           0            0   
1                   0    ...                  1           0            0   
2                   0    ...                  1           0            0   
3                   0    ...                  1           0            0   
4                   0    ...                  1           0            0   

   Noise.very_loud  beer_and_wine  full_bar  no alcohol  wifi.free  wifi.no  \
0                1              0         1           0          0        1   
1                0              0         1           0          0        1   
2                0              0         1           0          0        1   
3                0              0         1           0          0        1   
4                0              0         1           0          0        1   

   wifi.paid  
0          0  
1          0  
2          0  
3          0  
4          0  

[5 rows x 320 columns]

In [8]:
star_1.head(25)

business_id               review_id  date.year  date.month  \
9    VaN93_qJshvhx_wAfOM7eA  AI16r2fdd0bXD2K1EOV1Kw       2014           8   
25   hzUOqxpcnYYBgLlrgvjQYQ  182OqfzB6Epv64OMpAo3_w       2014           9   
27   qy7NtAyd7SXXPM0H44XVAg  GcqrLUckM9h8DnaAMZnIbw       2012           9   
31   1621ir5mjVgbHwxCbMAEjg  q8ijr7bauPp1zgTBvcKrbQ       2013           9   
36   tyC0LaPJo3ct_0JEkTwZxw  K_mrrAv2UzUgqqK-0eClxw       2014          12   
42   ke3RFq3mHEAoJE_kkRNhiQ  9q-b4iF4hV8vmAdMLPjRkg       2014           3   
65   p1ntTC9-CI6x5Y_nUoo6Ag  _cckT5V3Hxmoazejaby71Q       2013          12   
77   IW-NYIHk50mXV5Nz7nSGyw  hi7xTG6V1REhA3dmqVhYzg       2014           5   
79   KramfptRWKyQOm672d-w4Q  WA-cHl-qabrWZ8amai-J6w       2013           5   
90   HQ8N_EwJA3NdF1UApsF-fA  bzuMqdk28X4k2uS3hUvsgg       2011           2   
..                      ...                     ...        ...         ...   
163  jEWfxxA_kjeJE_Z_ku3zLA  T3sj0vi51Szoev5lORbVpg       2011           1   
169  T4ox3wUzFjWvyrjaTRoBbA  6hnfcnn_vwl5oS6KQ2jupQ       2013           6   
179  jEWfxxA_kjeJE_Z_ku3zLA  fLF8o4kzXgL2fppsG3KZYg       2011           2   
180  pfW2F6DibbJ3z9tYmEwgsw  vrm31uQJDzLfYOQQHa701g       2011          11   
212  6BXkb7kOhzk9cV7s09CkEQ  t53wJlJyitPu5GVsVq57Jg       2014           4   
221  8o-NLKy_XfbJtqljX9XLCA  wsoCrjABd815-LSPRNbGsQ       2013           3   
228  2a4wMeWkUcS2Qlx47_iQ-Q  rw0FIJecIyUecztjiw0g0w       2014           3   
250  winRNt7prallDbpaDMS9Ig  5WzJ3-lhFjQphx8FFTqcYQ       2014           7   
260  wftoNIA0Q8nX-Da47psBCA  aIyG0w5xqXuky_kKWOu88A       2013           3   
262  Y_Y91oeUrIwYa18F8ey5NQ  -vnzhFsgwrMYdJszD-JSpQ       2014          12   

     date.day  stars                                               text  \
9          19      1  Last few times I've came here after work, (abo...   
25         28      1  This placement doesn't even deserve a star so ...   
27         11      1  Horrible... I ordered car side to go because I...   
31         17      1  My friends and I all got the drunken noodle. I...   
36         31      1  "What is in the chicken burrito?"\n\n"Chicken....   
42         16      1  I was not impressed with the service or qualit...   
65          8      1  I cannot belive this place is still opened! Pa...   
77          8      1  I'd rather go through life uncircumcised than ...   
79         17      1  I was super turned off by the dishonesty of ou...   
90          7      1  They should change the name to Streets of Beji...   
..        ...    ...                                                ...   
163        29      1  Ra Sushi, you are so BLAH to me. We came to Te...   
169        20      1  Besides being one of the better ramen shops in...   
179        19      1  You know when you see someone and they look re...   
180        13      1  We used to really like this place and have tak...   
212        27      1  I'm shocked at the amount of stars being used ...   
221        27      1  My son and I were downtown and I decided to in...   
228        23      1  Visiting Arizona with our travel ball team. In...   
250        31      1  I love Grimaldi's pizza. But I will not come t...   
260        24      1  Overpriced and dirty.  Had lunch here with my ...   
262        30      1  Rude staff closed the door in my face at 9:59....   

     review_length  votes.cool  votes.funny  votes.useful  \
9              709           0            0             1   
25            1278           0            0             2   
27            1263           0            0             0   
31             209           0            0             0   
36             412           0            0             0   
42             352           0            0             2   
65             167           0            0             0   
77            1165           0            0             0   
79            1424           0            0             0   
90             695           1        

### Turn review topic_id_50 and topic_prob_50 columns into features

In [28]:


def turn_dist_feature(df):
    d1 = []
    for val in df['word_dist'].values:
        n = []
        for t in val:
            n.append(tuple(reversed(t)))
        d1.append(dict(n))
    return d1

d1 = turn_dist_feature(star_1)
d2 = turn_dist_feature(star_2)
d3 = turn_dist_feature(star_3)
d4 = turn_dist_feature(star_4)
d5 = turn_dist_feature(star_5)

d = d1+d2+d3+d4+d5
word_features = pd.DataFrame(d)
word_features.fillna(0, inplace=True)
word_features.head()


50  airport  amazing  area    asked  atmosphere  away  awesome      bad  \
0   0        0        0     0  2.6e-03           0     0        0  0.0e+00   
1   0        0        0     0  0.0e+00           0     0        0  0.0e+00   
2   0        0        0     0  0.0e+00           0     0        0  0.0e+00   
3   0        0        0     0  2.2e-03           0     0        0  0.0e+00   
4   0        0        0     0  0.0e+00           0     0        0  3.0e-03   

   bar ...   wait  walked  want  wasn  way     went  wife  wine  wings  yum  
0    0 ...      0       0     0     0    0  3.1e-03     0     0      0    0  
1    0 ...      0       0     0     0    0  0.0e+00     0     0      0    0  
2    0 ...      0       0     0     0    0  0.0e+00     0     0      0    0  
3    0 ...      0       0     0     0    0  2.5e-03     0     0      0    0  
4    0 ...      0       0     0     0    0  3.0e-03     0     0      0    0  

[5 rows x 164 columns]

In [29]:
star_1.reset_index(inplace=True, drop=True)
star_2.reset_index(inplace=True, drop=True)
star_3.reset_index(inplace=True, drop=True)
star_4.reset_index(inplace=True, drop=True)
star_5.reset_index(inplace=True, drop=True)

all_reviews = star_1.append([star_2, star_3, star_4, star_5])
all_reviews

business_id               review_id  date.year  date.month  \
0     VaN93_qJshvhx_wAfOM7eA  AI16r2fdd0bXD2K1EOV1Kw       2014           8   
1     hzUOqxpcnYYBgLlrgvjQYQ  182OqfzB6Epv64OMpAo3_w       2014           9   
2     qy7NtAyd7SXXPM0H44XVAg  GcqrLUckM9h8DnaAMZnIbw       2012           9   
3     1621ir5mjVgbHwxCbMAEjg  q8ijr7bauPp1zgTBvcKrbQ       2013           9   
4     tyC0LaPJo3ct_0JEkTwZxw  K_mrrAv2UzUgqqK-0eClxw       2014          12   
5     ke3RFq3mHEAoJE_kkRNhiQ  9q-b4iF4hV8vmAdMLPjRkg       2014           3   
6     p1ntTC9-CI6x5Y_nUoo6Ag  _cckT5V3Hxmoazejaby71Q       2013          12   
7     IW-NYIHk50mXV5Nz7nSGyw  hi7xTG6V1REhA3dmqVhYzg       2014           5   
8     KramfptRWKyQOm672d-w4Q  WA-cHl-qabrWZ8amai-J6w       2013           5   
9     HQ8N_EwJA3NdF1UApsF-fA  bzuMqdk28X4k2uS3hUvsgg       2011           2   
...                      ...                     ...        ...         ...   
3501  vvA3fbps4F9nGlAEYKk_sA  Amw0Y6bKlhOUxw7Cy1vM2g       2014          10   
3502  NzmTnhKa17kZbuc-M0u41w  EnSnjkduEHSb8zfdNdUbtQ       2011          11   
3503  zxjMiqlfUuIQRltSuj9FpQ  HdR9m9BafYSQo75FrHj8qw       2014           8   
3504  2bdKR3l4o-S1CscLqqnvVw  4t8qdFnxaDy2TuivniAyVg       2010          12   
3505  lk-6NXfpSgvNw7zm0xtUVw  LwIknaXelqi7Gwtibh1xxQ       2013           6   
3506  8MKeIqMlo6gTqBqXC0sj3A  ayYvjU_-iSTwqmV_d_RQHw       2014           7   
3507  svx5Gw-FJ6ZU9hb58aXgLg  HoeazfZb8QTRhDVqw83xKw       2014           8   
3508  S3FhtwXuiGMlGdBNtmiiSQ  ARZBtXdr_NbhVjR_waLPnA       2014           8   
3509  W8AYXV_V63ke_p1SE44-Cw  0Mvycfc8alUFPds5vWlD1A       2012           4   
3510  vcfQI3Tq8c5_nBimuwdNtg  YcVUzL9p-81rea-BKdAiLQ       2009          11   

      date.day  stars                                               text  \
0           19      1  Last few times I've came here after work, (abo...   
1           28      1  This placement doesn't even deserve a star so ...   
2           11      1  Horrible... I ordered car side to go because I...   
3           17      1  My friends and I all got the drunken noodle. I...   
4           31      1  "What is in the chicken burrito?"\n\n"Chicken....   
5           16      1  I was not impressed with the service or qualit...   
6            8      1  I cannot belive this place is still opened! Pa...   
7            8      1  I'd rather go through life uncircumcised than ...   
8           17      1  I was super turned off by the dishonesty of ou...   
9            7      1  They should change the name to Streets of Beji...   
...        ...    ...                                                ...   
3501         3      5  A+++ would do business with again. How come no...   
3502        22      5  Barros is my favorite pizza joint in town. We ...   
3503         1      5  Jhon gave us great customer service he saw us ...   
3504        15      5  What a great find! Located off of Yelp after a...   
3505        11      5  This is one of those "local joints" you see on...   
3506         3      5  My husband and I had a very enjoyable lunch. T...   
3507        18      5  This place never let's me down. Great food and...   
3508         3      5  Great great great customer service! They remem...   
3509        29      5  We eat at Sunrise frequently and have never be...   
3510         8      5  Ah, the go-to place for day drinking and ditch...   

      review_length  votes.cool  votes.funny  votes.useful  \
0               709           0            0             1   
1              1278           0            0             2   
2              1263           0            0             0   
3               209           0            0             0   
4               412           0            0             0   
5               352           0            0             2   
6               167           0            0             0   
7              1165           0            0             0   
8              1424           0            0   

In [36]:
word_features.reset_index(inplace=True, drop=True)
all_reviews.reset_index(inplace=True, drop=True)

In [37]:
df = pd.concat([all_reviews,word_features], axis=1)
df

business_id               review_id  date.year  date.month  \
0     VaN93_qJshvhx_wAfOM7eA  AI16r2fdd0bXD2K1EOV1Kw       2014           8   
1     hzUOqxpcnYYBgLlrgvjQYQ  182OqfzB6Epv64OMpAo3_w       2014           9   
2     qy7NtAyd7SXXPM0H44XVAg  GcqrLUckM9h8DnaAMZnIbw       2012           9   
3     1621ir5mjVgbHwxCbMAEjg  q8ijr7bauPp1zgTBvcKrbQ       2013           9   
4     tyC0LaPJo3ct_0JEkTwZxw  K_mrrAv2UzUgqqK-0eClxw       2014          12   
5     ke3RFq3mHEAoJE_kkRNhiQ  9q-b4iF4hV8vmAdMLPjRkg       2014           3   
6     p1ntTC9-CI6x5Y_nUoo6Ag  _cckT5V3Hxmoazejaby71Q       2013          12   
7     IW-NYIHk50mXV5Nz7nSGyw  hi7xTG6V1REhA3dmqVhYzg       2014           5   
8     KramfptRWKyQOm672d-w4Q  WA-cHl-qabrWZ8amai-J6w       2013           5   
9     HQ8N_EwJA3NdF1UApsF-fA  bzuMqdk28X4k2uS3hUvsgg       2011           2   
...                      ...                     ...        ...         ...   
9990  vvA3fbps4F9nGlAEYKk_sA  Amw0Y6bKlhOUxw7Cy1vM2g       2014          10   
9991  NzmTnhKa17kZbuc-M0u41w  EnSnjkduEHSb8zfdNdUbtQ       2011          11   
9992  zxjMiqlfUuIQRltSuj9FpQ  HdR9m9BafYSQo75FrHj8qw       2014           8   
9993  2bdKR3l4o-S1CscLqqnvVw  4t8qdFnxaDy2TuivniAyVg       2010          12   
9994  lk-6NXfpSgvNw7zm0xtUVw  LwIknaXelqi7Gwtibh1xxQ       2013           6   
9995  8MKeIqMlo6gTqBqXC0sj3A  ayYvjU_-iSTwqmV_d_RQHw       2014           7   
9996  svx5Gw-FJ6ZU9hb58aXgLg  HoeazfZb8QTRhDVqw83xKw       2014           8   
9997  S3FhtwXuiGMlGdBNtmiiSQ  ARZBtXdr_NbhVjR_waLPnA       2014           8   
9998  W8AYXV_V63ke_p1SE44-Cw  0Mvycfc8alUFPds5vWlD1A       2012           4   
9999  vcfQI3Tq8c5_nBimuwdNtg  YcVUzL9p-81rea-BKdAiLQ       2009          11   

      date.day  stars                                               text  \
0           19      1  Last few times I've came here after work, (abo...   
1           28      1  This placement doesn't even deserve a star so ...   
2           11      1  Horrible... I ordered car side to go because I...   
3           17      1  My friends and I all got the drunken noodle. I...   
4           31      1  "What is in the chicken burrito?"\n\n"Chicken....   
5           16      1  I was not impressed with the service or qualit...   
6            8      1  I cannot belive this place is still opened! Pa...   
7            8      1  I'd rather go through life uncircumcised than ...   
8           17      1  I was super turned off by the dishonesty of ou...   
9            7      1  They should change the name to Streets of Beji...   
...        ...    ...                                                ...   
9990         3      5  A+++ would do business with again. How come no...   
9991        22      5  Barros is my favorite pizza joint in town. We ...   
9992         1      5  Jhon gave us great customer service he saw us ...   
9993        15      5  What a great find! Located off of Yelp after a...   
9994        11      5  This is one of those "local joints" you see on...   
9995         3      5  My husband and I had a very enjoyable lunch. T...   
9996        18      5  This place never let's me down. Great food and...   
9997         3      5  Great great great customer service! They remem...   
9998        29      5  We eat at Sunrise frequently and have never be...   
9999         8      5  Ah, the go-to place for day drinking and ditch...   

      review_length  votes.cool  votes.funny ...      wait   walked  want  \
0               709           0            0 ...   0.0e+00  0.0e+00     0   
1              1278           0            0 ...   0.0e+00  0.0e+00     0   
2              1263           0            0 ...   0.0e+00  0.0e+00     0   
3               209           0            0 ...   0.0e+00  0.0e+00     0   
4               412           0            0 ...   0.0e+00  0.0e+00     0   
5               352           0            0 ...   0.0e+00  0.0e+00     0   
6               167           0            0 ...   0.0e+00  0.0e+0

In [38]:
for col in df.columns:
    print col
    print df[col].dtype

business_id
object
review_id
object
date.year
int64
date.month
int64
date.day
int64
stars
int64
text
object
review_length
int64
votes.cool
int64
votes.funny
int64
votes.useful
int64
user_id
object
topic_id
int64
topic_prob
float64
word_dist
object
50
float64
airport
float64
amazing
float64
area
float64
asked
float64
atmosphere
float64
away
float64
awesome
float64
bad
float64
bar
float64
bbq
float64
beef
float64
best
float64
better
float64
big
float64
bland
float64
blown
float64
bourbon
float64
bread
float64
bread pudding
float64
breakfast
float64
breakfast burrito
float64
burger
float64
burgers
float64
burrito
float64
came
float64
cheese
float64
chicago
float64
chicken
float64
chinese
float64
coffee
float64
come
float64
crust
float64
definitely
float64
delicious
float64
did
float64
didn
float64
dinner
float64
don
float64
donuts
float64
dos
float64
downtown phoenix
float64
eat
float64
eggs
float64
everytime
float64
excellent
float64
experience
float64
favorite
float64
fish
float64
flavo

In [9]:
# def change_to_int(df, col_names):
#     for name in col_names:
#         df[name] = df[name].astype(int)
#     return

# cols = ['stars', 'votes.cool', 'votes.funny', 'votes.useful', 'review_length']
# change_to_int(reviews, cols)
# reviews.dtypes

business_id      object
review_id        object
date.year        object
date.month       object
date.day         object
stars             int64
text             object
review_length     int64
...
topic_42    float64
topic_43    float64
topic_44    float64
topic_45    float64
topic_46    float64
topic_47    float64
topic_48    float64
topic_49    float64
Length: 65, dtype: object

## Join the topic distribution to business data
#### Note: At this point, we have 3,358 restaurants and 10,000 reviews from which I will build my models.

In [39]:
reviews = df
grouped_reviews = reviews.groupby('business_id').agg([np.sum, np.mean])
grouped_reviews.reset_index(inplace=True)

In [40]:
grouped_reviews.head()

business_id date.year         date.month      date.day        \
                                sum    mean        sum mean      sum  mean   
0  --5jkZ3-nUPZxUvtcbr8Uw      2014  2014.0          6  6.0       13  13.0   
1  -1bOb2izeJBZjHC7NWxiPA      2010  2010.0          8  8.0       26  26.0   
2  -2gyHPSL8pJjkSe5Ozzw6Q      6035  2011.7         18  6.0       30  10.0   
3  -3WVw1TNQbPBzaKCaQQ1AQ     18109  2012.1         53  5.9      172  19.1   
4  -3frEL8Vo8kyq4kfgTn0GA      2013  2013.0          2  2.0       11  11.0   

  stars      review_length ...  went      wife      wine      wings      yum  \
    sum mean           sum ...   sum mean  sum mean  sum mean   sum mean sum   
0     5  5.0            44 ...     0    0    0    0    0    0     0    0   0   
1     3  3.0           393 ...     0    0    0    0    0    0     0    0   0   
2    11  3.7          1088 ...     0    0    0    0    0    0     0    0   0   
3    36  4.0          5046 ...     0    0    0    0    0    0     0    0   0   
4     3  3.0           315 ...     0    0    0    0    0    0     0    0   0   

        
  mean  
0    0  
1    0  
2    0  
3    0  
4    0  

[5 rows x 349 columns]

In [19]:
grouped_reviews.columns

MultiIndex(levels=[[u'stars', u'review_length', u'votes.cool', u'votes.funny', u'votes.useful', u'topic_id_50', u'topic_prob_50', u'topic_0', u'topic_1', u'topic_2', u'topic_3', u'topic_4', u'topic_5', u'topic_6', u'topic_7', u'topic_8', u'topic_9', u'topic_10', u'topic_11', u'topic_12', u'topic_13', u'topic_14', u'topic_15', u'topic_16', u'topic_17', u'topic_18', u'topic_19', u'topic_20', u'topic_21', u'topic_22', u'topic_23', u'topic_24', u'topic_25', u'topic_26', u'topic_27', u'topic_28', u'topic_29', u'topic_30', u'topic_31', u'topic_32', u'topic_33', u'topic_34', u'topic_35', u'topic_36', u'topic_37', u'topic_38', u'topic_39', u'topic_40', u'topic_41', u'topic_42', u'topic_43', u'topic_44', u'topic_45', u'topic_46', u'topic_47', u'topic_48', u'topic_49', u'business_id'], [u'sum', u'mean', u'']],
           labels=[[57, 0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 17, 17, 18, 18, 19, 19, 20, 20, 21, 21, 22, 22, 

In [41]:
review_means = grouped_reviews.xs('mean', level=1, axis=1)
review_means['business_id'] = grouped_reviews['business_id'].values
review_means

/Users/aprilsong/anaconda/lib/python2.7/site-packages/IPython/kernel/__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from IPython.kernel.zmq import kernelapp as app


date.year  date.month  date.day  stars  review_length  votes.cool  \
0        2014.0         6.0      13.0    5.0           44.0         0.0   
1        2010.0         8.0      26.0    3.0          393.0         0.0   
2        2011.7         6.0      10.0    3.7          362.7         0.0   
3        2012.1         5.9      19.1    4.0          560.7         0.3   
4        2013.0         2.0      11.0    3.0          315.0         3.0   
5        2013.0         7.0      23.0    3.7         1195.0         0.0   
6        2012.5        11.0      17.0    4.5          151.5         0.5   
7        2014.0        11.0      28.0    2.0         1643.0         0.0   
8        2013.0         7.0      26.0    5.0          440.0         1.0   
9        2014.0         4.0      27.0    5.0          123.0         0.0   
...         ...         ...       ...    ...            ...         ...   
3348     2013.0         5.3      15.7    4.3          301.3         0.0   
3349     2012.8         7.8      16.0    2.8          526.8         0.0   
3350     2012.1         6.6      18.7    3.7          280.9         0.0   
3351     2012.6         8.0      19.9    4.6          509.5         0.0   
3352     2010.5         3.5       6.5    5.0          483.0         0.0   
3353     2014.0         5.5      28.5    4.5          426.5         0.5   
3354     2011.0         6.0       7.5    4.0          508.0         0.0   
3355     2014.0         8.0       1.0    5.0          276.0         0.0   
3356     2013.2         6.0      20.8    3.5          271.0         0.5   
3357     2013.5        12.0      16.0    4.5          267.0         0.0   

      votes.funny  votes.useful  topic_id  topic_prob           ...            \
0             0.0           0.0      23.0         0.7           ...             
1             0.0           0.0      28.0         1.0           ...             
2             0.0           0.7      19.7         0.8           ...             
3             0.4           0.6      30.1         0.7           ...             
4             1.0           2.0       1.0         1.0           ...             
5             0.0           0.0      33.3         0.8           ...             
6             0.0           0.5      36.0         0.7           ...             
7             0.0           0.0       6.0         1.0           ...             
8             1.0           1.0      44.0         0.7           ...             
9             1.0           1.0      10.0         0.7           ...             
...           ...           ...       ...         ...           ...             
3348          0.0           0.0      15.3         1.0           ...             
3349          0.0           0.2      33.8         1.0           ...             
3350          0.0           0.4      16.9         0.8           ...             
3351          0.0           0.2      24.6         0.9           ...             
3352          0.0           0.0      24.5         1.0           ...             
3353          0.5           0.5      25.0         0.9           ...             
3354          0.0           0.0      20.0         0.5           ...             
3355          0.0           0.0       2.0         0.6           ...             
3356          0.0           0.0      16.0         0.9           ...             
3357          0.0           0.0      36.5         0.9           ...             

      walked  want  wasn  way  went     wife     wine    wings  yum  \
0          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
1          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
2          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
3          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
4          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
5          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
6          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
7   

In [14]:
review_means.columns = map(lambda x : 'avg_' + x, review_means.columns.values)
review_means

avg_stars  avg_review_length  avg_votes.cool  avg_votes.funny  \
0           5.0               44.0             0.0              0.0   
1           3.0              393.0             0.0              0.0   
2           3.7              362.7             0.0              0.0   
3           4.0              560.7             0.3              0.4   
4           3.0              315.0             3.0              1.0   
5           3.7             1195.0             0.0              0.0   
6           4.5              151.5             0.5              0.0   
7           2.0             1643.0             0.0              0.0   
8           5.0              440.0             1.0              1.0   
9           5.0              123.0             0.0              1.0   
...         ...                ...             ...              ...   
3348        4.3              301.3             0.0              0.0   
3349        2.8              526.8             0.0              0.0   
3350        3.7              280.9             0.0              0.0   
3351        4.6              509.5             0.0              0.0   
3352        5.0              483.0             0.0              0.0   
3353        4.5              426.5             0.5              0.5   
3354        4.0              508.0             0.0              0.0   
3355        5.0              276.0             0.0              0.0   
3356        3.5              271.0             0.5              0.0   
3357        4.5              267.0             0.0              0.0   

      avg_votes.useful  avg_topic_id_50  avg_topic_prob_50  avg_topic_0  \
0                  0.0             41.0                0.6            0   
1                  0.0             44.0                0.5            0   
2                  0.7             37.0                0.4            0   
3                  0.6             28.1                0.6            0   
4                  2.0             18.0                0.2            0   
5                  0.0             42.7                0.4            0   
6                  0.5             40.5                0.9            0   
7                  0.0             37.0                0.4            0   
8                  1.0             33.0                0.6            0   
9                  1.0             35.0                0.2            0   
...                ...              ...                ...          ...   
3348               0.0             41.7                0.7            0   
3349               0.2             18.5                0.4            0   
3350               0.4             29.7                0.5            0   
3351               0.2             20.4                0.6            0   
3352               0.0              4.0                0.6            0   
3353               0.5             32.0                0.5            0   
3354               0.0             31.5                0.8            0   
3355               0.0             39.0                0.8            0   
3356               0.0             23.2                0.6            0   
3357               0.0             33.0                0.5            0   

      avg_topic_1  avg_topic_2           ...            avg_topic_41  \
0               0          0.0           ...                     0.6   
1               0          0.0           ...                     0.0   
2               0          0.0           ...                     0.0   
3               0          0.0           ...                     0.0   
4               0          0.0           ...                     0.0   
5               0          0.0           ...                     0.0   
6               0          0.0           ...                     0.5   
7               0          0.0           ...                     0.0   
8               0          0.0           ...                     0.0   
9               0          0.0           ...                     0.0   


In [42]:
restaurants

business_id  open  review_count  stars  cat.Active Life  \
0     EmzaQR5hQlF0WIl24NxAZA     1            23    2.5                0   
1     KPoTixdjoJxSqRSEApSAGg     1           322    4.0                0   
2     e5kc0CQ4R-PCCDgb274gSg     1           191    3.5                0   
3     45puCRQ6Vh_IIAy7kkfFDQ     1           170    3.5                0   
4     BESmH-lmMm34_EefL8X9fQ     1           204    4.0                0   
5     D09yG6Z3gcsh24Qn7Y4gYA     1            73    3.0                0   
6     kaIue7GRCmkPzDeHDBTttQ     1           152    4.0                0   
7     j8nyAaN_w5t6wYv-ni23sg     0            77    3.0                0   
8     IceZ4BtTz76eppjNWm-c6A     0            59    3.5                0   
9     -lOSaCuBRAvX5JBifx-EMw     1           105    3.0                0   
...                      ...   ...           ...    ...              ...   
3348  WlPYm5aK-ewwWeO4TlrF1g     1             7    5.0                0   
3349  mMJVKgRNlsr0YrkUBGqfQw     1            37    3.5                0   
3350  d7b78XiwyCApbKbWkuCGuw     1            36    3.0                0   
3351  ETrtpthOGuC8q7rE17s30g     1            55    3.5                0   
3352  u5cRdcNN-9V8SlsfO_Kl7w     1            40    4.5                0   
3353  n9H3Ay_jMjhKjUNZWtaICA     1            19    4.0                0   
3354  hQBLiFMHKq7OvsaGCb9AZA     1            69    4.0                0   
3355  tc1kER4tWn3LI99Rk-NVZw     1            37    4.0                0   
3356  q4MTywvT1EFuWiFMLQ7PXA     1            19    4.5                0   
3357  qCPBS-m_4uDO0EgIYGtoxw     1            28    4.0                1   

      cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                           0           0            0             0   
1                           0           0            0             0   
2                           0           0            0             0   
3                           0           0            0             0   
4                           0           0            0             0   
5                           0           0            0             0   
6                           0           0            0             0   
7                           0           0            0             0   
8                           0           0            0             0   
9                           0           0            0             0   
...                       ...         ...          ...           ...   
3348                        0           0            0             0   
3349                        0           0            0             0   
3350                        0           0            0             0   
3351                        0           0            0             0   
3352                        0           0            0             0   
3353                        0           0            0             0   
3354                        0           0            0             0   
3355                        0           0            0             0   
3356                        0           0            0             0   
3357                        0           0            0             0   

      cat.American (New)    ...      Noise.average  Noise.loud  Noise.quiet  \
0                      1    ...                  0           0            0   
1                      0    ...                  1           0            0   
2                      0    ...                  1           0            0   
3                      0    ...                  1           0            0   
4                      0    ...                  1           0            0   
5                      0    ...                  0           0            1   
6                      0    ...                  1           0            0   
7                      1    ...                  1           0            0   
8                      0    ...                 

In [48]:
review_means

date.year  date.month  date.day  stars  review_length  votes.cool  \
0        2014.0         6.0      13.0    5.0           44.0         0.0   
1        2010.0         8.0      26.0    3.0          393.0         0.0   
2        2011.7         6.0      10.0    3.7          362.7         0.0   
3        2012.1         5.9      19.1    4.0          560.7         0.3   
4        2013.0         2.0      11.0    3.0          315.0         3.0   
5        2013.0         7.0      23.0    3.7         1195.0         0.0   
6        2012.5        11.0      17.0    4.5          151.5         0.5   
7        2014.0        11.0      28.0    2.0         1643.0         0.0   
8        2013.0         7.0      26.0    5.0          440.0         1.0   
9        2014.0         4.0      27.0    5.0          123.0         0.0   
...         ...         ...       ...    ...            ...         ...   
3348     2013.0         5.3      15.7    4.3          301.3         0.0   
3349     2012.8         7.8      16.0    2.8          526.8         0.0   
3350     2012.1         6.6      18.7    3.7          280.9         0.0   
3351     2012.6         8.0      19.9    4.6          509.5         0.0   
3352     2010.5         3.5       6.5    5.0          483.0         0.0   
3353     2014.0         5.5      28.5    4.5          426.5         0.5   
3354     2011.0         6.0       7.5    4.0          508.0         0.0   
3355     2014.0         8.0       1.0    5.0          276.0         0.0   
3356     2013.2         6.0      20.8    3.5          271.0         0.5   
3357     2013.5        12.0      16.0    4.5          267.0         0.0   

      votes.funny  votes.useful  topic_id  topic_prob           ...            \
0             0.0           0.0      23.0         0.7           ...             
1             0.0           0.0      28.0         1.0           ...             
2             0.0           0.7      19.7         0.8           ...             
3             0.4           0.6      30.1         0.7           ...             
4             1.0           2.0       1.0         1.0           ...             
5             0.0           0.0      33.3         0.8           ...             
6             0.0           0.5      36.0         0.7           ...             
7             0.0           0.0       6.0         1.0           ...             
8             1.0           1.0      44.0         0.7           ...             
9             1.0           1.0      10.0         0.7           ...             
...           ...           ...       ...         ...           ...             
3348          0.0           0.0      15.3         1.0           ...             
3349          0.0           0.2      33.8         1.0           ...             
3350          0.0           0.4      16.9         0.8           ...             
3351          0.0           0.2      24.6         0.9           ...             
3352          0.0           0.0      24.5         1.0           ...             
3353          0.5           0.5      25.0         0.9           ...             
3354          0.0           0.0      20.0         0.5           ...             
3355          0.0           0.0       2.0         0.6           ...             
3356          0.0           0.0      16.0         0.9           ...             
3357          0.0           0.0      36.5         0.9           ...             

      walked  want  wasn  way  went     wife     wine    wings  yum  \
0          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
1          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
2          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
3          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
4          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
5          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
6          0     0     0    0     0  0.0e+00  0.0e+00  0.0e+00    0   
7   

In [46]:
restaurants = pd.merge(left=restaurants, right=review_means, how='inner', left_on='business_id', right_on='business_id')
restaurants.to_pickle('by_stars/restaurant_with_topic_dist1.pkl')
restaurants


business_id  open  review_count  stars_x  cat.Active Life  \
0     EmzaQR5hQlF0WIl24NxAZA     1            23      2.5                0   
1     KPoTixdjoJxSqRSEApSAGg     1           322      4.0                0   
2     e5kc0CQ4R-PCCDgb274gSg     1           191      3.5                0   
3     45puCRQ6Vh_IIAy7kkfFDQ     1           170      3.5                0   
4     BESmH-lmMm34_EefL8X9fQ     1           204      4.0                0   
5     D09yG6Z3gcsh24Qn7Y4gYA     1            73      3.0                0   
6     kaIue7GRCmkPzDeHDBTttQ     1           152      4.0                0   
7     j8nyAaN_w5t6wYv-ni23sg     0            77      3.0                0   
8     IceZ4BtTz76eppjNWm-c6A     0            59      3.5                0   
9     -lOSaCuBRAvX5JBifx-EMw     1           105      3.0                0   
...                      ...   ...           ...      ...              ...   
3348  WlPYm5aK-ewwWeO4TlrF1g     1             7      5.0                0   
3349  mMJVKgRNlsr0YrkUBGqfQw     1            37      3.5                0   
3350  d7b78XiwyCApbKbWkuCGuw     1            36      3.0                0   
3351  ETrtpthOGuC8q7rE17s30g     1            55      3.5                0   
3352  u5cRdcNN-9V8SlsfO_Kl7w     1            40      4.5                0   
3353  n9H3Ay_jMjhKjUNZWtaICA     1            19      4.0                0   
3354  hQBLiFMHKq7OvsaGCb9AZA     1            69      4.0                0   
3355  tc1kER4tWn3LI99Rk-NVZw     1            37      4.0                0   
3356  q4MTywvT1EFuWiFMLQ7PXA     1            19      4.5                0   
3357  qCPBS-m_4uDO0EgIYGtoxw     1            28      4.0                1   

      cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                           0           0            0             0   
1                           0           0            0             0   
2                           0           0            0             0   
3                           0           0            0             0   
4                           0           0            0             0   
5                           0           0            0             0   
6                           0           0            0             0   
7                           0           0            0             0   
8                           0           0            0             0   
9                           0           0            0             0   
...                       ...         ...          ...           ...   
3348                        0           0            0             0   
3349                        0           0            0             0   
3350                        0           0            0             0   
3351                        0           0            0             0   
3352                        0           0            0             0   
3353                        0           0            0             0   
3354                        0           0            0             0   
3355                        0           0            0             0   
3356                        0           0            0             0   
3357                        0           0            0             0   

      cat.American (New)  ...    wait_y  walked_y  want_y   wasn_y    way_y  \
0                      1  ...         0         0       0  0.0e+00  1.8e-03   
1                      0  ...         0         0       0  0.0e+00  0.0e+00   
2                      0  ...         0         0       0  0.0e+00  0.0e+00   
3                      0  ...         0         0       0  0.0e+00  0.0e+00   
4                      0  ...         0         0       0  2.8e-04  0.0e+00   
5                      0  ...         0         0       0  0.0e+00  0.0e+00   
6                      0  ...         0         0       0  0.0e+00  0.0e+00   
7                      1  ...         0         0       0  0.0e+00  0.0e+00   
8   

### Create initial feature matrix

In [47]:
print restaurants.columns.values

[u'business_id' u'open' u'review_count' u'stars_x' u'cat.Active Life'
 u'cat.Adult Entertainment' u'cat.Afghan' u'cat.African' u'cat.Airports'
 u'cat.American (New)' u'cat.American (Traditional)' u'cat.Amusement Parks'
 u'cat.Appliances' u'cat.Arabian' u'cat.Arcades' u'cat.Argentine'
 u'cat.Art Galleries' u'cat.Arts & Crafts' u'cat.Arts & Entertainment'
 u'cat.Asian Fusion' u'cat.Australian' u'cat.Auto Repair' u'cat.Automotive'
 u'cat.Bagels' u'cat.Bakeries' u'cat.Bangladeshi' u'cat.Barbeque'
 u'cat.Bars' u'cat.Basque' u'cat.Beauty & Spas' u'cat.Bed & Breakfast'
 u'cat.Beer Bar' u'cat.Beer Gardens' u'cat.Beer, Wine & Spirits'
 u'cat.Belgian' u'cat.Bistros' u'cat.Bowling' u'cat.Brasseries'
 u'cat.Brazilian' u'cat.Breakfast & Brunch' u'cat.Breweries' u'cat.British'
 u'cat.Bubble Tea' u'cat.Buffets' u'cat.Burgers' u'cat.Burmese'
 u'cat.Butcher' u'cat.Cafes' u'cat.Cafeteria' u'cat.Cajun/Creole'
 u'cat.Cambodian' u'cat.Candy Stores' u'cat.Cantonese' u'cat.Car Wash'
 u'cat.Caribbean' u'cat.C

In [49]:
restaurants = restaurants.loc[:, :'yum_x']
restaurants


business_id  open  review_count  stars_x  cat.Active Life  \
0     EmzaQR5hQlF0WIl24NxAZA     1            23      2.5                0   
1     KPoTixdjoJxSqRSEApSAGg     1           322      4.0                0   
2     e5kc0CQ4R-PCCDgb274gSg     1           191      3.5                0   
3     45puCRQ6Vh_IIAy7kkfFDQ     1           170      3.5                0   
4     BESmH-lmMm34_EefL8X9fQ     1           204      4.0                0   
5     D09yG6Z3gcsh24Qn7Y4gYA     1            73      3.0                0   
6     kaIue7GRCmkPzDeHDBTttQ     1           152      4.0                0   
7     j8nyAaN_w5t6wYv-ni23sg     0            77      3.0                0   
8     IceZ4BtTz76eppjNWm-c6A     0            59      3.5                0   
9     -lOSaCuBRAvX5JBifx-EMw     1           105      3.0                0   
...                      ...   ...           ...      ...              ...   
3348  WlPYm5aK-ewwWeO4TlrF1g     1             7      5.0                0   
3349  mMJVKgRNlsr0YrkUBGqfQw     1            37      3.5                0   
3350  d7b78XiwyCApbKbWkuCGuw     1            36      3.0                0   
3351  ETrtpthOGuC8q7rE17s30g     1            55      3.5                0   
3352  u5cRdcNN-9V8SlsfO_Kl7w     1            40      4.5                0   
3353  n9H3Ay_jMjhKjUNZWtaICA     1            19      4.0                0   
3354  hQBLiFMHKq7OvsaGCb9AZA     1            69      4.0                0   
3355  tc1kER4tWn3LI99Rk-NVZw     1            37      4.0                0   
3356  q4MTywvT1EFuWiFMLQ7PXA     1            19      4.5                0   
3357  qCPBS-m_4uDO0EgIYGtoxw     1            28      4.0                1   

      cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                           0           0            0             0   
1                           0           0            0             0   
2                           0           0            0             0   
3                           0           0            0             0   
4                           0           0            0             0   
5                           0           0            0             0   
6                           0           0            0             0   
7                           0           0            0             0   
8                           0           0            0             0   
9                           0           0            0             0   
...                       ...         ...          ...           ...   
3348                        0           0            0             0   
3349                        0           0            0             0   
3350                        0           0            0             0   
3351                        0           0            0             0   
3352                        0           0            0             0   
3353                        0           0            0             0   
3354                        0           0            0             0   
3355                        0           0            0             0   
3356                        0           0            0             0   
3357                        0           0            0             0   

      cat.American (New)  ...    wait_x  walked_x  want_x   wasn_x    way_x  \
0                      1  ...         0         0       0  0.0e+00  1.8e-03   
1                      0  ...         0         0       0  0.0e+00  0.0e+00   
2                      0  ...         0         0       0  0.0e+00  0.0e+00   
3                      0  ...         0         0       0  0.0e+00  0.0e+00   
4                      0  ...         0         0       0  2.8e-04  0.0e+00   
5                      0  ...         0         0       0  0.0e+00  0.0e+00   
6                      0  ...         0         0       0  0.0e+00  0.0e+00   
7                      1  ...         0         0       0  0.0e+00  0.0e+00   
8   

In [51]:
restaurants.columns.values

array([u'business_id', u'open', u'review_count', u'stars_x',
       u'cat.Active Life', u'cat.Adult Entertainment', u'cat.Afghan',
       u'cat.African', u'cat.Airports', u'cat.American (New)',
       u'cat.American (Traditional)', u'cat.Amusement Parks',
       u'cat.Appliances', u'cat.Arabian', u'cat.Arcades', u'cat.Argentine',
       u'cat.Art Galleries', u'cat.Arts & Crafts',
       u'cat.Arts & Entertainment', u'cat.Asian Fusion', u'cat.Australian',
       u'cat.Auto Repair', u'cat.Automotive', u'cat.Bagels',
       u'cat.Bakeries', u'cat.Bangladeshi', u'cat.Barbeque', u'cat.Bars',
       u'cat.Basque', u'cat.Beauty & Spas', u'cat.Bed & Breakfast',
       u'cat.Beer Bar', u'cat.Beer Gardens', u'cat.Beer, Wine & Spirits',
       u'cat.Belgian', u'cat.Bistros', u'cat.Bowling', u'cat.Brasseries',
       u'cat.Brazilian', u'cat.Breakfast & Brunch', u'cat.Breweries',
       u'cat.British', u'cat.Bubble Tea', u'cat.Buffets', u'cat.Burgers',
       u'cat.Burmese', u'cat.Butcher', u'cat.C

In [179]:
restaurants.to_pickle('restaurant/restaurant_with_topic_dist.pkl')

In [52]:
del restaurants['date.year_x']
del restaurants['date.month_x']
del restaurants['stars_y']
del restaurants['topic_id_x']
del restaurants['topic_prob_x']


In [53]:
restaurants.to_pickle('by_stars/restaurant_with_topic_dist1.pkl')

In [ ]:
def run_and_score(model, X, y,round_method):
    y_labels = y.astype(str)
    
    if model == Ridge:
        mod = model(alpha=.5)
    else:
        mod = model()
    if model == Ridge or model == LinearRegression:
        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=12)
        mod.fit(X_train, y_train)
        print 'Score as linear regression problem:'
        get_rss(mod, X_test, y_test)
        print
        get_cross_val_score(mod, X, y)
        print '-------------------------------------------'
        print
        print 'Score after converting to multiclass problem:'
        predicted_labels = map(lambda x : str(round_method(x)), mod.predict(X_test))
        if round_method == cust_round_half:
            y_test_labels = map(lambda x: str(x), y_test)
        else:
            y_test_labels = map(lambda x: str(round_method(x)), y_test)
        print metrics.accuracy_score(y_test_labels, predicted_labels)
        print
        print metrics.classification_report(y_test_labels, predicted_labels)
        print
        categories=np.unique(y_test_labels)
        print pd.DataFrame(confusion_matrix(y_test_labels, predicted_labels),
             index = pd.MultiIndex.from_product([['actual'], categories]),
             columns = pd.MultiIndex.from_product([['predicted'], categories]))
    else:
        y_labels = map(lambda x: str(round_method(x)), y)
        categories = np.unique(y_labels)
        X_train, X_test, y_train, y_test = train_test_split(X, y_labels, test_size=0.2, random_state=12)
        mod.fit(X_train, y_train)
        get_cross_val_score(mod, X, y_labels)
        predicted = mod.predict(X_test)
        print
        print metrics.classification_report(y_test, predicted)
        print
        print pd.DataFrame(confusion_matrix(y_test, predicted),
             index = pd.MultiIndex.from_product([['actual'], categories]),
             columns = pd.MultiIndex.from_product([['predicted'], categories]))
    
def run_all_models(X, y):
    print '======================LINEAR REGRESSION======================'
    print 'HALF INTEGER ROUND'
    print
    run_and_score(LinearRegression, X, y, cust_round_half)
    print 
    print 'INTEGER ROUND'
    run_and_score(LinearRegression, X, y, cust_round)
    print
    print '======================RIDGE REGRESSION======================'
    print 'HALF INTEGER ROUND'
    print
    run_and_score(LinearRegression, X, y, cust_round_half)
    print 
    print 'INTEGER ROUND'
    run_and_score(LinearRegression, X, y, cust_round)
    print
    print '======================SVC CLASSIFIER======================'
    print 'HALF INTEGER ROUND'
    print
    run_and_score(svm.SVC, X, y, cust_round_half)
    print 
    print 'INTEGER ROUND'
    run_and_score(svm.SVC, X, y, cust_round)

In [ ]:
X1_1 = X1.loc[:,:'wifi.paid']

run_and_score(LinearRegression, X1_1, y, cust_round_half)

run_and_score(LinearRegression, X1, y, cust_round_half)

df = DataFrame( , index=['Including Half Stars', 'Excluding Half Stars']
               
               )

arrays = [np.array(['Linear Regression', 'Linear Regression', 'Ridge Regression', 'Ridge Regression','SVC', 'SVC']),
         np.array(['Half Integer Round', 'Integer Round', 'Half Integer Round', 'Integer Round', 'Half Integer Round', 'Integer Round'])]

X1_scores = [0.38244047619, 0.642857142857, 0.391369047619, 0.645833333333, 0.331737, 0.656944]

pd.DataFrame(X1_scores, index=arrays)
[0.416666666667, 0.223326481628, 0.394345238095, 0.622023809524, 0.351986, 0.661705] 


[0.360119047619, 0.639880952381, ,0.360908,0.661703]


In [54]:
a = [ 1.5,  2. ,  2.5,  3. ,  3.5,  4. ,  4.5,  5. ]
map(lambda

'[1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]'

In [58]:
b = [ 1.5,  2. ,  3.5,  4. ,  4.5,  5. ]
c = sorted(list(set(a+b)))
c

[1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]

In [ ]:
pcax1 = [0.360119047619, 0.639880952381, 0.360119047619, 0.639880952381, 0.360908, 0.661703]
pcax2 = [0.360119047619, 0.639880952381, 0.360119047619, 0.639880952381, 0.360908, 0.661703]
lowvar1 = [0.388392857143, 0.598214285714, 0.388392857143, 0.598214285714, 0.326361, 0.650988]
lowvar2 = [0.395833333333, 0.605654761905, 0.395833333333, 0.605654761905, 0.336499, 0.656340]
scaled1 = [0.386904761905, 0.616071428571, 0.386904761905, 0.616071428571, 0.385890, 0.678077]
scaled2 = [0.404761904762, 0.614583333333, 0.404761904762, 0.614583333333, 0.410044, 0.678077]
[0.386904761905, 0.619047619048, 0.386904761905, 0.619047619048, 0.369865, 0.662301]